test/integration/apiserver/apply/apply_test.go

```diff
@@ -830,3 +830,59 @@ func TestApplyConvertsManagedFieldsVersion(t *testing.T) {
 		t.Fatalf("expected:\n%v\nbut got:\n%v", expected, actual)
 	}
 }
+
+// TestClearManagedFields verifies it's possible to clear the managedFields
+func TestClearManagedFields(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	_, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		Namespace("default").
+		Resource("configmaps").
+		Name("test-cm").
+		Param("fieldManager", "apply_test").
+		Body([]byte(`{
+			"apiVersion": "v1",
+			"kind": "ConfigMap",
+			"metadata": {
+				"name": "test-cm",
+				"namespace": "default",
+				"labels": {
+					"test-label": "test"
+				}
+			},
+			"data": {
+				"key": "value"
+			}
+		}`)).
+		Do().
+		Get()
+	if err != nil {
+		t.Fatalf("Failed to create object using Apply patch: %v", err)
+	}
+
+	_, err = client.CoreV1().RESTClient().Patch(types.MergePatchType).
+		Namespace("default").
+		Resource("configmaps").
+		Name("test-cm").
+		Body([]byte(`{"metadata":{"managedFields": [{}]}}`)).Do().Get()
```

Comment:
Do you need the empty dictionary in the lisit, or would this also work: `{"metadata":{"managedFields": []}}`
And what about: `{"metadata":{"managedFields": null}}`

Reply:
I haven't tried null, good point.
Using an empty array does not work as we have a check in the FieldManager to ensure we don't reset on an empty array.
It's in the update function, I can post a link tomorrow